Introduce a module providing rastermasks read from external files

jenshannoschwalm opened the pull request:
Had been requested a few times, one was #12551

1. In preferences we have a root folder containing the mask pfm files.
2. Select a file to be used as a raster mask, please note the file is kept in history. Only .pfm files are supported
3. Chosing the RGB components of that file to be used for the rastermask via 'mode'
4. If the module has focus it visualizes the later used raster mask.
5. OpenCL code is provided (except for the visualizing mode)

As this is maybe the first darktable module using the lately introduces per-module-instance self->data there is some how-to-use information provided. Here we use it to avoid reading the pfm file again and again as that would be pretty costly.

For demo purpose here is pfm file (unzip it)
[treebark.zip](https://github.com/user-attachments/files/20026454/treebark.zip)
and a screenshot
![Bildschirmfoto vom 2025-05-04 09-01-40](https://github.com/user-attachments/assets/b2b7f03f-364f-48d3-b114-8612677fe491)

jenshannoschwalm commented on src/iop/rasterfile.c:
```diff
@@ -0,0 +1,705 @@
+  gtk_widget_set_name(button, "non-flat");
+  gtk_widget_set_tooltip_text(button, _("select a pfm file to be advertized as a raster mask,\n"
+      "CAUTION: path must be set in preferences/processing before choosing"));
+  gtk_box_pack_start(GTK_BOX(hbox), button, FALSE, FALSE, 0);
```
good and done.

jenshannoschwalm commented:
Some code deduplications and refinements in module.
Also updated the gtk code as suggested.
While being here also changed the gtk code of lut3d (i took the file handler stuff from there).

jenshannoschwalm commented:
Right about proposed refactoring changes ,👍 will do

But otherwise you think the module it worth to be included?

Donatzsky commented:
Many smartphones embed a depth-map in the raw files. Would it be possible for this module to also load those?

As for the name, I think "raster mask import" would be better.

jenshannoschwalm commented:
> As for the name, I think "raster mask import" would be better.

good.

jenshannoschwalm commented:
> I think it should, all modules in dt when changing some values are automatically activated.

Right, fixed that in latest force-push

Release note:
Added a new module "raster mask import". After setting up a root folder containing pfm files, a file can be selected to be used as a raster mask, it is scaled to full image size. You can also define what combination of the RGB channels of that pfm file will be used for the raster mask.
